### ISSUE TITLE
How to set style

### ISSUE THREAD

**Issue by anfractuosity** [gtk-rs/gtk#287]
I notice the following has been commented out:
```
    //pub fn add_provider_for_screen<T: IsA</*Ignored*/StyleProvider>>(screen: &gdk::Screen, provider: &T, priority: u32) {
    //    unsafe { TODO: call ffi::gtk_style_context_add_provider_for_screen() }
    //}
```

Which is what I used to initiate the use of a CSS style before, Im just wondering what the preferred method 
is now?

cheers
Chris

**Comment by anfractuosity** [comment]
Thanks to you guys' help on IRC!, this is what I have so far:

```
extern crate gtk_sys;
...
use glib::translate::*;
unsafe {
gtk_style_context_add_provider_for_screen(WindowExt::get_screen(&window).unwrap().to_glib_none().0,css,gtk::STYLE_PROVIDER_PRIORITY_APPLICATION as u32);
}
```

The issue now is that css, needs to be in the form of GtkStyleProvider

cheers!

**Comment by gkoz** [comment]
Assuming `css` is `&CssProvider` I think the workaround would look something like this. 
 
```rust
unsafe {
    let screen = WindowExt::get_screen(&window).unwrap().to_glib_none();
    let css: Stash<*mut gtk_ffi::GtkCssProvider, _> = css.to_glib_none();
    gtk_style_context_add_provider_for_screen(screen.0,
        // We know `CssProvider` implements `StyleProvider` so just cast the pointer
        css.0 as *mut _,
        gtk::STYLE_PROVIDER_PRIORITY_APPLICATION as u32);
}
```

I'm variables to make this more readable but you never assign `to_glib_none().0` to a variable, only the whole stash and then take the pointer out of it when needed.

**Comment by anfractuosity** [comment]
Thank-you so much!
It works :)

I just had to create the variable 'w' because I had a lifetime issue.

```
unsafe {
    let w = WindowExt::get_screen(&window).unwrap();
    let screen = w.to_glib_none();
    let css: Stash<*mut gtk_sys::GtkCssProvider, _> = css.to_glib_none();
    gtk_style_context_add_provider_for_screen(screen.0,
        // We know `CssProvider` implements `StyleProvider` so just cast the pointer
        css.0 as *mut _,
        gtk::STYLE_PROVIDER_PRIORITY_APPLICATION as u32);
}
```
Thanks!

**Comment by gkoz** [comment]
I think this has been fixed in #288
